Fixing the issue when omi user profile does not exist on system

sarojcare opened the pull request:
This change set includes
1. Different way to handle the error cases in pbuild and product deployment environment.
2. Calling new exception handler to print the exception appropriately.
eg.
"Calling getpwnam() with user name parameter "omi", returned an error with errno = 1(Operation not permitted )"

msftclas commented:
@sarojcare,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

sarojcare commented:
The tavis failure is due to "SCXErrnoUserException" declaration in pal is not visible. 

I have tried changing travis.yml so that it can be available. But no luck. I think it will be fixed automatically once we check in changes in pal branch.

sarojcare commented:
Temporarily fixed Travis to pass all checks. Will revert back before check-in.

jeffaco commented on source/code/shared/tools/scx_ssl_config/scx_ssl_config.cpp:
```diff
@@ -471,8 +471,12 @@ static int DoGenerate(const wstring & targetPath, int startDays, int endDays,
-           throw SCXCoreLib::SCXErrnoException(L"getpwnam", errno, SCXSRCLOCATION);
+        else if(errno !=0 && errno !=ENOENT && errno !=ESRCH) {
+            const char* isTestEnv = getenv("SCX_TESTRUN_ACTIVE");
+            if(isTestEnv)
```
Suggest just throwing an exception if `! isTestEnv`. Realize that it's perfectly normal on build systems (and DEV systems) to not have an OMI account.

If you want to do more, I'd suggest that you just log the error at 477 (via logging, which I think is enabled here) rather than use wcout, which will clutter the unit test output.

sarojcare commented:
> Do be sure to revert change to .travis.yml. I'm not actually 100% sure why you needed that anyway, care to enlighten me?

I had added the exceptions in pal branch and had decided so as all the exceptions used in scx_ssl_config.cpp are from the scxexception.cpp file in pal. 

In travis.yml it is defined to use the 'pal' master branch due to which exception class added was not seen during compiling of scx_ssl_config.cpp. Due to this travis check was failing. Hence I have modified the travis.yml  to checkout appropriate branch instead of master branch so that compilation will pass.

Now as you suggested I have moved the exception class added in pal to opsmgr and reverted back the changes in travis.
